core/ipc/rpc/kafka/src/main/java/org/opennms/core/ipc/rpc/kafka/KafkaRpcClientFactory.java

```diff
@@ -136,12 +148,17 @@
                     // The request is for the current location, invoke it directly
                     return module.execute(request);
                 }
+                Span span = tracer.buildSpan(module.getId()).start();
                 final JmsQueueNameFactory topicNameFactory = new JmsQueueNameFactory(
                         KafkaRpcConstants.RPC_REQUEST_TOPIC_NAME, module.getId(), request.getLocation());
                 String requestTopic = topicNameFactory.getName();
                 String marshalRequest = module.marshalRequest(request);
                 // Generate RPC Id for every request to track request/response.
                 String rpcId = UUID.randomUUID().toString();
+                span.setTag("rpcId", rpcId);
```

Comment:
We may want to move these tag name to constants somewhere in `core/tracing` so that other RPC impl. can use the same ones too.

Reply:
Can we add the location as a tag too?